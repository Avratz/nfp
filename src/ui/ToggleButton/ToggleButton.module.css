.toggleButton {
  color: var(--neutral-4);
  font-family: var(--font-dmSans);
  font-weight: var(--font-bold);
  font-size: var(--text-sm);
  padding: 0.3rem 0.75rem 0.25rem;
  display: inline-block;
}

.checked {
  color: var(--neutral-2);
  background-color: var(--neutral-8);
  border-radius: 5rem;
}
