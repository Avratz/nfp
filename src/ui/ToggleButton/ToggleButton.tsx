import { RadioGroup } from '@headlessui/react'
import { cx } from 'class-variance-authority'

import styles from './ToggleButton.module.css'

type ToggleButtonProps = {
  value: string
  children: React.ReactNode
}

export function ToggleButton({ value, children }: ToggleButtonProps) {
  return (
    <RadioGroup.Option value={value}>
      {({ checked }) => (
        <span
          className={cx(styles.toggleButton, checked ? styles.checked : '')}
        >
          {children}
        </span>
      )}
    </RadioGroup.Option>
  )
}
