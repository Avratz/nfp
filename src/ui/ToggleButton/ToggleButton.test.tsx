import { RadioGroup } from '@headlessui/react'
import { render } from '@testing-library/react'
import { ToggleButton } from './ToggleButton'

describe('ToggleButton', () => {
  it('renders default', () => {
    const { container } = render(
      <RadioGroup>
        <ToggleButton value="ToggleButton">ToggleButton</ToggleButton>
      </RadioGroup>,
    )
    expect(container).toMatchSnapshot()
  })
})
