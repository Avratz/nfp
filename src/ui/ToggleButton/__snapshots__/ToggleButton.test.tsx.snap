// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders default 1`] = `
<div>
  <div
    id="headlessui-radiogroup-:r0:"
    role="radiogroup"
  >
    <div
      aria-checked="false"
      class="toggleButton"
      data-headlessui-state=""
      id="headlessui-radiogroup-option-:r1:"
      role="radio"
      tabindex="0"
    >
      ToggleButton
    </div>
  </div>
</div>
`;
